kcidb/monitor/test_output.py

```diff
@@ -85,3 +85,89 @@ def test_min(self):
         self.assertEqual('utf-8', html.get_content_charset())
         content = html.get_payload()
         self.assertIn("We detected a new revision!", content)
+
+    def test_subject_and_body_length(self):
+        """Check subject and body length of Notification """
+
+        db_client = db.Client("sqlite::memory:")
+        db_client.init()
+        oo_client = oo.Client(db_client)
+        db_client.load({
+            "version": self.version,
+            "checkouts": [
+                {
+                    "contacts": [
+                        "rdma-dev-team@redhat.com"
+                    ],
+                    "start_time": "2020-03-02T15:16:15.790000+00:00",
+                    "git_repository_branch": "wip/jgg-for-next",
+                    "git_commit_hash": "5e29d1443c46b6ca70a4c940a67e8c09f05dcb7e",
+                    "patchset_hash": "",
+                    "git_repository_url": "https://git.kernel.org/pub/scm/linux/kernel/git/arm64/linux.git",
+                    "misc": {
+                        "pipeline_id": 467715
+                    },
+                    "id": "origin:1",
+                    "origin": "origin",
+                    "patchset_files": [],
+                    "valid": True,
+                },
+            ],
+            "builds": [
+                                {
+                    "id": "kernelci:kernelci.org:619fecfc6d932a49b5f2efb0",
+                    "checkout_id": "origin:1",
+                    "origin": "origin",
+                    "comment": "v5.16-rc2-19-g383a44aec91c",
+                    "start_time": "2021-11-25T20:07:24.499000+00:00",
+                    "duration": 486.695294857,
+                    "architecture": "x86_64",
+                    "config_name": "x86_64_defconfig+x86-chromebook+kselftest",
+                    "valid": True
+                },
+            ]
+        })
+        revision = oo_client.query(orm.Pattern.parse(">revision#"))["revision"][0]
+        message = NotificationMessage(
+            ["kernelci-results-staging@groups.io"],
+            subject="JINJA2_TEMPLATE_FOR_YOUR_SUBJECT",
+            body="JINJA2_TEMPLATE_FOR_YOUR_BODY"
+        )
+        notification = Notification(revision, "test_subscription", message)
+        message = notification.render()
+        self.assertIsInstance(message, email.message.EmailMessage)
+        self.assertEqual(len(message.get("Subject", "")), 32)
+        self.assertEqual(len(message.get_body('plain').get_content()[:-1]), 29)
+
+        empty_message = NotificationMessage(
+            ["kernelci-results-staging@groups.io"],
+            subject="", body="")
+        empty_notification = Notification(revision, "test_subscription_2", empty_message)
+        empty_message = empty_notification.render()
+        self.assertIsInstance(empty_message, email.message.EmailMessage)
```

Comment:
I don't think we need to check that `render()` returns an `EmailMessage` in every test. We could save a few lines of code, if we do it only once.